Scripts/Navigation/NavigableScrollElement.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System.Collections;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.UI;
+
+namespace DUCK.DebugMenu.Navigation
+{
+	public class NavigableScrollElement : MonoBehaviour, ISelectHandler
+	{
+		private static NavigableScrollElement tweeningElement;
+
+		[SerializeField]
+		private float scrollSpeed = 1000f;
+
+		private ScrollRect scrollRect;
+		private RectTransform rootCanvasTransform;
+		private RectTransform scrollRectTransform;
+		private float scrollRectTop;
+		private float scrollRectBottom;
+
+		private void Start()
+		{
+			scrollRect = GetComponentInParent<ScrollRect>();
+			rootCanvasTransform = (RectTransform) GetComponentInParent<Canvas>().transform;
+			scrollRectTransform = ((RectTransform) scrollRect.transform);
+
+			// get world space corners of the scroll rect
+			Vector3[] corners = new Vector3[4];
+			scrollRectTransform.GetWorldCorners(corners);
+
+			// now we know the min/max y values of the scroll rect in "canvas space"
+			scrollRectTop = rootCanvasTransform.InverseTransformPoint(corners[1]).y;
+			scrollRectBottom = rootCanvasTransform.InverseTransformPoint(corners[0]).y;
+		}
+
+		public void OnSelect(BaseEventData eventData)
+		{
+			// work out if this element is fully visible to the scroll view
+
+			// now get world space corners of our own rect
+			Vector3[] corners = new Vector3[4];
+			((RectTransform)transform).GetWorldCorners(corners);
```

Comment:
Above there are two explicit casts with a space after, but here there is no space. 
Should be consistent.